### PR TITLE
Detect Fuse on Karaf

### LINKED ISSUE #372: Research Fuse-on-Karaf fingerprints
Read about JBoss Fuse and make a list of fingerprints. Update the fingerprinting doc with that information.

Email JBoss folks to take a look at the fingerprints, see if they agree.

### LINKED ISSUE #480: Detect Fuse on Karaf
Need to detect JBoss Fuse running on Apache Karaf.

### PR DESCRIPTION
**Pull request by noahl**
Scan running processes and use the locate command to detect JBoss Fuse
running on Apache Karaf.

As a bonus, also add a fact 'jboss.fuse.init-files' which looks for
init files that contain the string 'fuse'. It has some false
positives, but could be useful for finding systems that probably have
fuse.

Closes #480 . Closes #372 .

### CODE REVIEW

File: rho_playbook.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@
     - dmi
     - etc_release
     - file_contents
+    - fuse_on_karaf
```
Comment (chambridge) [inline rho_playbook.yml]: I would prefer we keep the jboss work together and call the role `jboss_fuse_on_karaf` placing it towards the end with the `jboss_eap` role

**Review by chambridge — COMMENTED** [review]
Comment on role name, although its more about grouping with the other middleware roles.

You need to generate the latest facts doc.

Also I think updates are likely needed to:
https://github.com/quipucords/rho/blob/master/doc/source/remote_programs.rst

File: rho_playbook.yml (hunk 1 above)
Comment (noahl) [reply rho_playbook.yml]: I moved everything to the `jboss_eap` role and updated the fact_docs and `remote_programs.rst`. Thank you!

File: rho_playbook.yml [hunk 2]
```diff
@@ -16,7 +16,6 @@
     - dmi
     - etc_release
     - file_contents
-    - fuse_on_karaf
```
Comment (chambridge) [inline rho_playbook.yml]: Didn't mean for you to remove the role. Just rename it and have it located with `jboss_eap` in the playbook.
Example
```
- jboss_eap
- jboss_eap_scan
- brms
- fuse
- jboss_fuse_on_karaf
```

File: roles/jboss_eap/tasks/main.yml [hunk 3]
```diff
@@ -115,3 +115,88 @@
       ignore_errors: yes
       become: true
       when: '"jboss.eap.init-files" in facts_to_collect'
+
+# Tasks that can locate a KARAF_HOME directory
```
Comment (chambridge) [inline roles/jboss_eap/tasks/main.yml]: Sorry for the confusion, just wanted the role to be run in a different order. Not combined with the eap role.

Comment (noahl) [reply roles/jboss_eap/tasks/main.yml]: My mistake! I just pushed another change which I think is what you are looking for.

**Review by chambridge — APPROVED** [review]

**Pull request merged**